packages/apps/api/src/app.ts

```diff
@@ -65,6 +65,7 @@ function registerUsersUserIdResource(app: Application): void {
   app.options('/users/:user_id', cors(corsOptions));
   app.get('/users/:user_id', cors(corsOptions), (request: Request, response: Response) => getUser(sierraClient, auth0Client, request, response));
   app.put('/users/:user_id', cors(corsOptions), (request: Request, response: Response) => updateUser(sierraClient, auth0Client, request, response));
+  app.post('/users/:user_id/auth', cors(corsOptions), (request: Request, response: Response) => authenticateUser(auth0Client, request, response));
```

Comment:
I wonder if this is the correct path for this?

Looking at `authenticateUser` in `user.ts` below, it looks like we expect both the user ID to be provided as a path parameter, and the email address and password as request parameters. The issue I think we _may_ run into in the future is if ourselves or someone else wants to be able to test a users credentials (email address and password) before they know what the users ID is. In this case, they won't be able to populate the `:user_id` path parameter. In general I don't think it makes sense to have to provide both user ID and email address anyway.

We have an existing endpoint `[POST] /auth` that currently tests Sierra credentials. This was originally going to be used by the Auth0 action scripts, but we've changed that so Auth0 goes directly to Sierra instead and as a result that endpoint is now unused.

How would you feel about switching the path here so that the Auth0 credential testing makes use of the existing [`POST] /auth` endpoint instead?

Reply:
As a note here too - the handler here would be registered in its own function since it's not a method operating on the `/user/:user_id` resource - see below for examples.

Reply:
Ah, I didn't realize there were other user endpoints already that weren't under here. I'll move that out.

+1 on updating to use /auth, doing it by userId doesn't really make sense anyway since the parameter is never actually used.